src/IO/GEHDF5Wrapper.cxx

```diff
@@ -419,94 +617,144 @@ Succeeded GEHDF5Wrapper::initialise_efficiency_factors(const std::string& path)
-    m_memspace_ptr= new H5::DataSpace(2, count.data());
+    // AB: this is only used for geo data, so for now lets ensure the file is sino. If its reused, we can change this. 
+    if(!is_geo_file())
+        error("File is Geometry. Aborting");
+
+    if(offset[0] != 0 || offset[1] != 0) //AB there are other C++ ways of doing this, but this is the most readable really.
+        error("Only {0,0} offset supported. Aborting");
+    if(stride[0] != 1 || stride[1] != 1)
+        error("Only {1,1} stride supported. Aborting");
+
+    // We know the size of the DataSpace
+    hsize_t str_dimsf[2] {m_NX_SUB, m_NY_SUB} ;
+    output.resize(m_NX_SUB*m_NY_SUB);
+
+    m_dataspace.selectHyperslab(H5S_SELECT_SET, str_dimsf, offset.data());
+    m_memspace_ptr= new H5::DataSpace(2, str_dimsf);
     m_dataset_sptr->read(output.get_data_ptr(), H5::PredType::NATIVE_UINT32, *m_memspace_ptr, m_dataspace);
     output.release_data_ptr();
 
-    //  // TODO error checking
     return Succeeded::yes;
 }
 
 //PW Developed for Efficiency Factors
-Succeeded GEHDF5Wrapper::get_from_2d_dataset(const std::array<unsigned long long int, 2>& offset,
-                                        const std::array<unsigned long long int, 2>& count,
-                                        const std::array<unsigned long long int, 2>& stride,
-                                        const std::array<unsigned long long int, 2>& block,
-                                        Array<1, float> &output)
+Succeeded GEHDF5Wrapper::get_from_2d_dataset(Array<1, float> &output,
+                                             const std::array<unsigned long long int, 2>& offset,
+                                             const std::array<unsigned long long int, 2>& stride)
+                                        
 {
-    m_dataspace.selectHyperslab(H5S_SELECT_SET, count.data(), offset.data());
-    m_memspace_ptr= new H5::DataSpace(2, count.data());
+    if(!is_norm_file())
+        error("The file provided is not norm data. Aborting");
+
+    if(offset[0] != 0 || offset[1] != 0) //AB there are other C++ ways of doing this, but this is the most readable really.
+        error("Only {0,0} offset supported. Aborting");
+    if(stride[0] != 1 || stride[1] != 1)
+        error("Only {1,1} stride supported. Aborting");
+
+    // We know the size of the DataSpace
+    hsize_t str_dimsf[2] {m_NX_SUB, m_NY_SUB} ;
+    output.resize(m_NX_SUB*m_NY_SUB);
+
+    m_dataspace.selectHyperslab(H5S_SELECT_SET, str_dimsf, offset.data());
+    m_memspace_ptr= new H5::DataSpace(2, str_dimsf);
     m_dataset_sptr->read(output.get_data_ptr(), H5::PredType::NATIVE_FLOAT, *m_memspace_ptr, m_dataspace);
     output.release_data_ptr();
 
-    //  // TODO error checking
     return Succeeded::yes;
 }
 
 //! \todo Array read as UINT32 should have an output of std::uint32_t.
 //! \todo Check the H5 data type as it's unlikely that it is NATIVE_UINT32, it may be STD_U32LE.
```

Comment:
remove this comment (and other similar ones). [Doc on `H5::DataSet::read`](http://davis.lbl.gov/Manuals/HDF5-1.6.5/cpplus_RM/classH5_1_1DataSet.html#a10) says
>void H5::DataSet::read (void * buf, const DataType & mem_type,...)
> This function reads raw data from this dataset into the buffer buf, converting from file datatype and dataspace to memory datatype mem_type and dataspace mem_space.

Therefore, it's correct that we want the memory type specify as a "native uint32"